reposerver/repository/repository.go

```diff
@@ -1372,3 +1372,56 @@ func (s *Service) GetHelmCharts(ctx context.Context, q *apiclient.HelmChartsRequ
 	}
 	return &res, nil
 }
+
+func (s *Service) TestRepository(ctx context.Context, q *apiclient.TestRepositoryRequest) (*apiclient.TestRepositoryResponse, error) {
+	repo := q.Repo
+	checks := map[string]func() error{
+		"git": func() error {
+			return s.TestRepo(repo)
```

Comment:
I believe we do not need a new `TestRepo()` method and that we can use `TestRepo()` from `util/git` package here.

I guess the intention behind this was to use repository locking, right? If yes, since the `git.TestRepo()` method uses purely in-memory storage (so, uses go-git), we won't require repository locking when testing for repository access.

Or was there another reason to not use `git.TestRepo()`?

Reply:
Yes, the intention was as you mentioned. I will use git.TestRepo() as we are not moving from go-git to fork/exec as part of this change.